Factory/DiactorosFactory.php

```
@@ -0,0 +1,159 @@
+            ->withParsedBody($symfonyRequest->request->all())
+        ;
+
+        foreach ($symfonyRequest->attributes->all() as $key => $value) {
+            $request = $request->withAttribute($key, $value);
+        }
+
+        return $request;
+    }
+
+    /**
+     * Converts Symfony uploaded files array to the PSR one.
+     *
+     * @param array $uploadedFiles
+     *
+     * @return array
+     */
+    private function getFiles(array $uploadedFiles)
+    {
+        $files = array();
+
+        foreach ($uploadedFiles as $key => $value) {
+            if ($value instanceof UploadedFile) {
+                $files[$key] = $this->createUploadedFile($value);
+            } else {
+                $files[$key] = $this->getFiles($value);
+            }
+        }
+
+        return $files;
+    }
+
+    /**
+     * Creates a PSR-7 UploadedFile instance from a Symfony one.
+     *
+     * @param UploadedFile $symfonyUploadedFile
+     *
+     * @return UploadedFileInterface
+     */
+    private function createUploadedFile(UploadedFile $symfonyUploadedFile)
+    {
+        return new DiactorosUploadedFile(
+            $symfonyUploadedFile->getRealPath(),
+            $symfonyUploadedFile->getSize(),
+            $symfonyUploadedFile->getError(),
+            $symfonyUploadedFile->getClientOriginalName(),
+            $symfonyUploadedFile->getClientMimeType()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createResponse(Response $symfonyResponse)
+    {
+        $stream = new DiactorosStream('php://temp', 'wb+');
+        if ($symfonyResponse instanceof StreamedResponse) {
+            ob_start(function ($buffer) use ($stream) {
+                $stream->write($buffer);
+
+                return false;
+            });
+
+            $symfonyResponse->sendContent();
+            ob_end_clean();
+        } else {
+            $stream->write($symfonyResponse->getContent());
```

The `BinaryFileResponse` won't work with this - but it'll just fail silently. Could we `fopen` the file and use that as the stream in `new DiactorosStream()` for `BinaryFileResponse`?